### PR TITLE
PCA and SVD for low-rank matrices, LOBPCG for positive-defined generalized eigenvalue problem - copy

### PR DESCRIPTION
**Pull request by pearu**
This is a copy of PR #29488 to help the merging process.

### CODE REVIEW

**Review by vincentqb — APPROVED** [review]
As mentioned [here](https://github.com/pytorch/pytorch/pull/29488#issuecomment-598826556), LGTM. Thanks!

**Comment by vincentqb** [comment]
Started internal test at the same time.

File: test/test_torch.py [hunk 1]
```diff
@@ -13501,6 +13644,319 @@ def test_eig(self, device, dtype):
...
+
+            # generalized eigenvalue problem, largest eigenvalues
+            E, V = lobpcg(A, B=B, k=k, n=n, largest=True)
+            self.assertEqual(matmul(A, V) / E.max(), mm(matmul(B, V), (E / E.max()).diag_embed()),
+                             prec=prec)
+
+    @skipCPUIfNoLapack
+    @onlyCPU
+    @dtypes(torch.double)
+    def test_lobpcg_torchscript(self, device, dtype):
```
Comment (ailzhang) [inline test/test_torch.py]: `test_torch.py` is only for eager frontend tests. All torchscript tests should go to `test_jit`

**Review by ailzhang — COMMENTED** [review]
Please move the test to proper place in a followup PR. 
Also @vincentqb please let me know that xla test is broken before it lands  ;)  thanks!

File: test/test_torch.py (hunk 1 above)
Comment (vincentqb) [reply test/test_torch.py]: Thanks for the clarification! See #34898.

**Pull request closed without merging**